src/sid/update_states.py

```diff
@@ -60,41 +66,24 @@ def update_states(
     if optional_state_columns is None:
         optional_state_columns = {"reason_for_infection": False, "contacts": False}
 
-    # Reduce all existing countdowns by 1.
-    for countdown in COUNTDOWNS:
-        states[countdown] -= 1
-
-    # Make changes where the countdown is zero.
-    for countdown, info in COUNTDOWNS.items():
-        locs = states.index[states[countdown] == 0]
-        for to_change, new_val in info.get("changes", {}).items():
-            states.loc[locs, to_change] = new_val
-
-        for new_countdown in info.get("starts", []):
-            states.loc[locs, new_countdown] = states.loc[locs, f"{new_countdown}_draws"]
-
-    states["newly_infected"] = newly_infected_contacts | newly_infected_events
-    states["immune"] = states["immune"] | states["newly_infected"]
-
-    if channel_infected_by_contact is not None:
-        states["channel_infected_by_contact"] = channel_infected_by_contact
-
-    if channel_infected_by_event is not None:
-        states["channel_infected_by_event"] = channel_infected_by_event
+    states = _update_countdowns(states)
 
-    # Update states with new infections and add corresponding countdowns.
-    locs = states.query("newly_infected").index
-    states.loc[locs, "ever_infected"] = True
-    states.loc[locs, "cd_ever_infected"] = 0
-    states.loc[locs, "cd_immune_false"] = states.loc[locs, "cd_immune_false_draws"]
-    states.loc[locs, "cd_infectious_true"] = states.loc[
-        locs, "cd_infectious_true_draws"
-    ]
+    states = _update_info_on_newly_infected(
+        states, newly_infected_contacts, newly_infected_events
+    )
 
     states = _kill_people_over_icu_limit(states, params, next(seed))
 
     # important: this has to be called after _kill_people_over_icu_limit!
-    states["newly_deceased"] = states["cd_dead_true"] == 0
+    states["newly_deceased"] = states.eval(IS_NEWLY_DECEASED)
+
+    if share_known_cases is not None:
+        to_be_processed_test = _compute_new_tests_with_share_known_cases(
+            states, share_known_cases
+        )
+
+    if to_be_processed_test is not None:
```

Comment:
I think if `share_known_cases` is `None` this would not be defined and would raise a `NameError`

Reply:
No, since `to_be_processed_tests` is also an argument of update_states.